pallets/providers/src/utils.rs

```diff
@@ -449,6 +449,13 @@ where
             Error::<T>::StorageStillInUse
         );
 
+        // Check that the sign off period since the BSP signed up has passed
+        ensure!(
+            frame_system::Pallet::<T>::block_number()
+                >= bsp.sign_up_block + T::BspSignUpLockPeriod::get(),
+            Error::<T>::SignOffPeriodNotPassed
+        );
+
```

Comment:
The lock for signing off is only checked for BSPs but the sign up block has been added to MSPs as well. Since we have this check for the former, shouldn't we have something similar for the latter?

Reply:
The reason for this lock is so that it is more expensive for an attacker to sign up a bunch of BSPs. There is no point in suddenly signing up a bunch of MSPs, and this kind of measure, increases the barrier of entry. Now you have to consider wether you're willing to lock these tokens for this amount of time, to venture into providing storage.

So that's why I didn't do the same for MSPs. I still record their signup to be more coherent and because it is somewhat useful information anyway.